pulp_2to3_migration/app/models/base.py

```diff
@@ -11,24 +19,164 @@ class MigrationPlan(Model):
         plan (models.JSONField): The migration plan in the JSON format
     """
     plan = JSONField()
+    _real_plan = None
+
+    @property
+    def plan_view(self):
+        """
+        Window to view the validated migration plan data through.
+        """
+        if not self._real_plan:
+            self._real_plan = _InternalMigrationPlan(self.plan)
+
+        return self._real_plan
+
+    def get_plugins(self):
+        """
+        Return a list of pulp2 plugins to migrate.
+        """
+        return self.plan_view.plugins_to_migrate
 
     def get_repositories(self):
         """
         Return a list of pulp2 repositories to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 repositories from the self.plan
-        return []
+        return self.plan_view.repositories_to_migrate
+
+    def get_pulp3_repository_setup(self):
+        """
+        Return a dict of pulp3 repositories to create and information about e.g. versions.
+        """
+        return self.plan_view.repositories_to_create
 
     def get_importers(self):
         """
         Return a list of pulp2 importers to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 importers from the self.plan
-        return []
+        return self.plan_view.importers_to_migrate
 
     def get_distributors(self):
         """
         Return a list of pulp2 distributors to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 distributors from the self.plan
-        return []
+        return self.plan_view.distributors_to_migrate
+
+    def get_missing_resources(self):
+        """
+        Return a dict of any resources listed in the plan but missing from Pulp 2.
+
+        Repositories and Importers are enumerated by repo_id, Distributors by distributor_id.
```

Comment:
Unless it would make sense to do distributors by distributor_id also.

Reply:
Did you mean by repo_id? I think it's fine as it is for now. We might change something when we migrate distributors.

Reply:
Yup, that's what I meant.  Thanks